src/result-list.js

```diff
@@ -12,22 +12,48 @@ function ResultList(retriever, options) {
   this._retriever = retriever;
   this._onFailure = options.onFailure;
   this._onSuccess = options.onSuccess;
-  this._page = options.page || 1;
+  this._page = options.page ? options.page - 1 : 0;
   this._perPage = options.perPage || 20;
+  this._requestPromises = [];
+
+  this._retriever.query({per_page: this._perPage});
 }
 
 ResultList.prototype.next = function() {
-  this._retriever.query({page: this._page, per_page: this._perPage});
   this._page += 1;
+
+  this._retriever.query({page: this._page});
+
+  var promise = this._requestPromises[this._page] ?
+    this._requestPromises[this._page] : this._get();
+
+  return promise;
+};
+
+ResultList.prototype.prev = function() {
```

Comment:
> can we spell this all the way out? confusing this with any of these seems like an expensive tradeoff for saving 4 characters

Sure, but I don't see how it's confusing. `prev` is a common abbreviation for `previous` in programming.